This is a patch that can be applied to the magic file used by file(1) to
recognize mono assemblies.
Apply it to the magic file (usually in /usr/share/file/magic or
/usr/share/misc/magic) and recompile it with file -C.

--- magic.old	2006-03-24 21:12:25.000000000 +0100
+++ magic	2006-03-24 21:12:17.000000000 +0100
@@ -7205,6 +7205,7 @@
 >>>>(0x3c.l+4)	leshort		0x290	PA-RISC
 >>>>(0x3c.l+22)	leshort&0x0100	>0	32-bit
 >>>>(0x3c.l+22)	leshort&0x1000	>0	system file
+>>>>(0x3c.l+232)	lelong	>0	Mono/.Net assembly
 
 >>>>(0x3c.l+0xf8)	string		UPX0 \b, UPX compressed
 >>>>(0x3c.l+0xf8)	search/0x140	PEC2 \b, PECompact2 compressed
